consensus/parlia/fusionfork_test.go

```diff
@@ -0,0 +1,146 @@
+package parlia
+
+import (
+	"container/heap"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func TestValidatorHeap(t *testing.T) {
+	testCases := []struct {
+		description  string
+		k            int
+		validators   []common.Address
+		votingPowers []uint64
+		expected     []common.Address
+	}{
+		{
+			description: "normal case",
+			k:           2,
+			validators: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+				common.HexToAddress("0x3"),
+			},
+			votingPowers: []uint64{
+				300,
+				200,
+				100,
+			},
+			expected: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+			},
+		},
+		{
+			description: "same voting power",
+			k:           2,
+			validators: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+				common.HexToAddress("0x3"),
+			},
+			votingPowers: []uint64{
+				300,
+				100,
+				100,
+			},
+			expected: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+			},
+		},
+		{
+			description: "zero voting power and k > len(validators)",
+			k:           5,
+			validators: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+				common.HexToAddress("0x3"),
+				common.HexToAddress("0x4"),
+			},
+			votingPowers: []uint64{
+				300,
+				0,
+				0,
+				0,
+			},
+			expected: []common.Address{
+				common.HexToAddress("0x1"),
+			},
+		},
+		{
+			description: "zero voting power and k < len(validators)",
+			k:           2,
+			validators: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+				common.HexToAddress("0x3"),
+				common.HexToAddress("0x4"),
+			},
+			votingPowers: []uint64{
+				300,
+				0,
+				0,
+				0,
+			},
+			expected: []common.Address{
+				common.HexToAddress("0x1"),
+			},
+		},
+		{
+			description: "all zero voting power",
+			k:           2,
+			validators: []common.Address{
+				common.HexToAddress("0x1"),
+				common.HexToAddress("0x2"),
+				common.HexToAddress("0x3"),
+				common.HexToAddress("0x4"),
+			},
+			votingPowers: []uint64{
+				0,
+				0,
+				0,
+				0,
+			},
+			expected: []common.Address{},
+		},
+	}
+	for _, tc := range testCases {
```

Comment:
This does not call the `getTopValidatorsByVotingPower` function, if we change `getTopValidatorsByVotingPower` later, the case here can still pass.

Reply:
Fixed